### PR TITLE
Add closable prop into side panel

### LINKED ISSUE #1294: Improve docs string for dialog events
#### Is your feature request related to a problem? Please describe
The current docs string for the ui.dialog events parameter is `events – The events to capture on this dialog`. However, there is a specific list of events that can be captured but this isn't documented making it hard for the user to know how to use this feature.

#### Describe the solution you'd like
Change the description to `events – The events to capture on this dialog. One of 'dismissed'.`

to match other descriptions like `ui.zone(direction)`:
"direction – Layout direction. One of 'row', 'column'. See enum h2o_wave.ui.ZoneDirection."

#### Notes
* Sidepanel and notification bar might also be related?

### PR DESCRIPTION
**Pull request by marek-mihok**
This PR introduces `closable` prop for side panel. It also updates API descriptions for `events` prop for both side panel and dialog. The updated APIs look as follows:

__Side panel:__

```ts
export interface SidePanel {
  /** The side panel's title. */
  title: S
  /** The components displayed in this side panel. */
  items: Component[]
  /** The width of the dialog, e.g. '400px'. Defaults to '600px'. */
  width?: S
  /** An identifying name for this component. */
  name?: Id
  /** The events to capture on this side panel. One of 'dismissed'. */
  events?: S[]
  /** True to prevent closing when clicking or tapping outside the side panel. Prevents interacting with the page behind the side panel. Defaults to False. */
  blocking?: B
  /** True if the side panel should have a closing 'X' button at the top right corner. */
  closable?: B
}
```

__Dialog:__
```ts
export interface Dialog {
  /** The dialog's title. */
  title: S
  /** The components displayed in this dialog. */
  items: Component[]
  /** The width of the dialog, e.g. '400px'. Defaults to '600px'. */
  width?: S
  /** True if the dialog should have a closing 'X' button at the top right corner. */
  closable?: B
  /** True to prevent closing when clicking or tapping outside the dialog. Prevents interacting with the page behind the dialog. Defaults to False. */
  blocking?: B
  /** Dialog with large header banner, mutually exclusive with `closable` prop. Defaults to False. */
  primary?: B
  /** An identifying name for this component. */
  name?: Id
  /** The events to capture on this dialog. One of 'dismissed'. */
  events?: S[]
}
```

It also brings updated docs to make dialog and side panel consistent with each other.

Closes #1161 
Closes #1294

### CODE REVIEW

**Review by mturoci — CHANGES_REQUESTED** [review]
Thanks @marek-mihok, a few minor comments.

File: ui/src/side_panel.tsx [hunk 1]
```diff
@@ -32,33 +32,36 @@ export interface SidePanel {
...
 export default bond(() => {
   const
     onDismiss = () => {
       const
-        { blocking, name, events } = sidePanelB() || {},
+        { closable, blocking, name, events } = sidePanelB() || {},
         ev = events?.find(e => e === 'dismissed')
 
-      if (!blocking && ev && name) wave.emit(name, ev, true)
+      if ((closable || !blocking) && ev && name) wave.emit(name, ev, true)
```
[the remark this answers is not in the document]
Comment (marek-mihok) [reply ui/src/side_panel.tsx]: You are right. I've changed this also for dialog and tested it.

File: ui/src/side_panel.tsx [hunk 2]
```diff
@@ -32,33 +32,36 @@ export interface SidePanel {
   items: Component[]
   /** The width of the dialog, e.g. '400px'. Defaults to '600px'. */
   width?: S
+  /** True to prevent closing when clicking or tapping outside the side panel. Prevents interacting with the page behind the side panel. Defaults to False. */
+  blocking?: B
```
Comment (mturoci) [inline ui/src/side_panel.tsx:36]: Preserve the existing attributes order and place any new optional attributes at the end.

**Review by mturoci — CHANGES_REQUESTED** [review]

**Review by mturoci — APPROVED** [review]
Thanks @marek-mihok, looking good.

**Pull request merged**